### PR TITLE
feat!: Widen CustomProperties type to map[string]interface{} to align with GitHub API

### PR DESCRIPTION
**Pull request by spalberg**
Fixes: #3229.

BREAKING CHANGE: `PushEventRepository.CustomProperties` is changed from `map[string]string` to `map[string]interface{}`.

adds tests for json unmarshalling of projects

### CODE REVIEW

File: github/repos_test.go
Comment (gmlewis) [inline github/repos_test.go:4509]: Instead of an actual response, how about if you just put one value of each type that you found, please?

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @spalberg !
Just one comment, then we should be able to merge after a second LGTM+Approval from any other contributor to this repo.

**Comment by gmlewis** [comment]
Maybe @chapurlatn can review these changes?

File: github/repos_test.go
Comment (spalberg) [reply github/repos_test.go]: I minimized the json as you suggested to have a small partial project

File: github/repos_test.go
Comment (gmlewis) [inline github/repos_test.go:4490]: How about this?

```suggestion
			data:           []byte(`{"custom_properties":{"boolean":false,"int":42,"key":"value","array":["a","b","c"]}}`),
```

Comment (spalberg) [reply github/repos_test.go:4490]: This would imply that there is a possible integer value when there is non (that we know of). I used the different custom property types offered by GitHub (see [here](https://docs.github.com/en/organizations/managing-organization-settings/managing-custom-properties-for-repositories-in-your-organization#:~:text=This%20can%20either%20be%20a%20text%20string%2C%20a%20single%20select%20field%2C%20a%20multi%20select%20field%2C%20or%20a%20true/false%20boolean)) and observed their representation in the api, using their label in the github ui as key for this json.

I guess it depends on whats the goal - testing that the unmarshalling works according to the type or testing the possible values that github could send for the given field. What do you think?

Comment (gmlewis) [reply github/repos_test.go:4490]: Oh, I see, well at least please change the boolean from a string to `false` without the quotes.

Comment (spalberg) [reply github/repos_test.go:4490]: GitHub returns the value for boolean type custom properties as strings ("true" or "false"). Should I remove the quotes regardless?

Comment (gmlewis) [reply github/repos_test.go:4490]: Are you sure?  We have hundreds of endpoints that return booleans, and none of them are strings.
What makes you say this?  You made a boolean custom property and it is returning a string?!?

Comment (spalberg) [reply github/repos_test.go:4490]: Yes, I made custom properties of all types and looked at the api response from github.
This is the actual response from the github api with one custom property (named accordingly) of each type configured:

```jsonc
{
  // ...
   "custom_properties": {
      "type-multi-select": [
        "a",
        "b"
      ],
      "type-single-select": "b",
      "type-text": "lala",
      "type-true-false": "true"
  },
  //...
}
```

Comment (spalberg) [reply github/repos_test.go:4490]: I'd love if they would return an actual boolean here :smile:

Comment (gmlewis) [reply github/repos_test.go:4490]: OK, bizarre.
Then it looks like there is one linting failure remaining... let me look.

Comment (gmlewis) [reply github/repos_test.go:4490]: It looks like the generated files are out-of-date if you could please update them.

Comment (spalberg) [reply github/repos_test.go:4490]: The whole integration of file generation is a really cool golang feature!
Just pushed the updated files :+1:

**Review by gmlewis — APPROVED** [review]
Thank you, @spalberg !
LGTM

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by arymoraes — APPROVED** [review]
LGTM

And yeah I remember when I was doing https://github.com/google/go-github/pull/3200 that a `boolean` type custom property comes as `string` from the GitHub API.

**Comment by gmlewis** [comment]
Thank you, @arymoraes !
Merging.

**Pull request merged**
